gluten-core/src/main/scala/org/apache/gluten/execution/FileSourceScanExecTransformer.scala

```diff
@@ -126,6 +127,15 @@ abstract class FileSourceScanExecTransformerBase(
     relation.location.inputFiles.toSeq
   }
 
+  override def getRootPathsInternal: Seq[String] = {
+    relation.location match {
+      case c: CatalogFileIndex =>
+        val index = c.filterPartitions(Nil)
```

Comment:
Why not use `org.apache.spark.sql.execution.datasources.CatalogFileIndex#rootPaths`? `c.filterPartitions(Nil)` seems to be a dangerous operation, it may list all hive partitions.

Reply:
I saw this function appear twice in the CatalogFileIndex, so I think this may not be an expensive call. Thanks for pointing this out, make sense to me.